rfcs/Subworlds.md

```diff
@@ -0,0 +1,139 @@
+
+## Motivation 🏃
+
+The primary motivation for using `Subworlds` to allows systems and queries to operate on disjoint sets of entities and their components.
+
+## Guide-level explanation 📝
+
+### Terms & Definitions 📚
+
+- `Subworld`: This refers to the original definition of a `World` in bevy 0.5. (essentially a collection of components which belong to a set of entities). 
+  - _Note_: I may use `World` and `Subworld` interchangeably.
+- `Worlds`: This can be thought of as a collection of `Subworlds`. 
+
+### Example ⭐
+- Your game is composed of levels. 
+- The player is currently on level 1 and nearing the end, so you would like to begin loading level 2.
+- Level 1 and 2 are completely disjoint and do not share any data (except perhaps the player themselves).
+- The difficulty resides in the fact that entities from levels 1 and 2 contain similar components, but the entities themselves are disjoint.
+  - In order to begin populating level 2's entities, something must exist to distinguish the two sets of entities, so that systems operating for level 1 (e.g. _physics_) do not operate on level 2's entities, even though they may contain physics components.
+- Only having a single `World` requires that systems filter the entities based on what level they belong to, adding overhead and complexity to either the `Query`, the system, the components describing the entities, or a combination.
+- `Subworlds` would provide a clean and efficient solution to this problem. 
+  - The primary systems `[physics, render, etc...]` would only operate on Level 1's `Subworld`.
+  - A `load_next_level` system could be running concurrently on Level 2's `Subworld`. 
+  - Then when Level 2 begins, the primary systems operate the on later `Subworld` while the former one gets saved/dropped.
+
+### Things `Subworlds` would allow 🙂
+- Running systems on subworld `A` on thread `A`, while running the same systems on subworld `B` on thread `B`.
+- Running systems on specific disjoint groups of entities based on some criteria.
+- Running systems on subworld `A` every frame and on subworld `B` every 10 seconds.
+- Helps systems stay clean and not require overuse of _marker-components_ (and possibly _relations_) which can lead to excessive archetypal fracturing which will end up hindering performance. 
+- Give developers the choice to use either a _single_, _global_, world that contains every entity (_note_: for some games, that might makes the most sense and be the most performing), or multiple `Subworlds` if they so desire.
```

Comment:
This is less of a benefit and more of a literal description? I'm not sure it fits well here :)